Add RIDs for MIRACLE LINUX 8 and 9

m10k opened the pull request:
Whenever we build dotnet RPMs for MIRACLE LINUX 8 (formerly Asianux Server 8) we have to
apply patches that add RIDs for MIRACLE LINUX. To streamline the build process, we would
like to add our RIDs to the upstream sources, in the same way CentOS and other RHEL-
derivatives do.

This commit adds the following runtime IDs:
 - "miraclelinux"
 - "miraclelinux-x64"
 - "miraclelinux.8"
 - "miraclelinux.8-x64"
 - "miraclelinux.9"
 - "miraclelinux.9-x64"

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=61304) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=61304) before we can accept your contribution.<br/><br/>:x: m10k [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=61304)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=61304) it.</sub>

m10k commented:
I have looked at the PRs that added RIDs for other distributions (Alpine 3.14 #55857, Debian 11 #56789, Fedora 36 #57832), and they don't seem any different. Since this PR adds a new distro, not just a new version of an existing distro, are there other files that need to be modified?

safern commented:
@m10k you want to add it to this file as well: https://github.com/dotnet/runtime/blob/ec15b53daa184583ef978dfd59e30ca4791e9470/src/libraries/Microsoft.NETCore.Platforms/src/runtimeGroups.props

Here is an example of a new platform addition: https://github.com/dotnet/runtime/commit/ec15b53daa184583ef978dfd59e30ca4791e9470

m10k commented:
@safern Thanks for the feedback! I had already added Miracle Linux to `runtimeGroups.props`, so that doesn't seem to be the issue. The PR for Rocky Linux introduced some other changes that I don't have in my PR, but they are not necessary for Miracle Linux, because on ML we don't have the minor version in /etc/os-release's VERSION_ID.

As far as I can tell, the issue is that the RID graph needs to be rebuilt (`/p:UpdateRuntimeFiles=true` needs to be passed), which is is something we also have to do when building dotnet on our buildserver - I don't know how I would do that here though.

safern commented:
Ah I see, then you have to run the build with `/p:UpdateRuntimeFiles=true` and then commit whatever changed into this PR.

m10k committed:
Add RIDs for MIRACLE LINUX 8 and 9

Whenever we build dotnet RPMs for MIRACLE LINUX 8 (formerly Asianux Server 8) we have to
apply patches that add RIDs for MIRACLE LINUX. To streamline the build process, we would
like to add our RIDs to the upstream sources, in the same way CentOS and other RHEL-
derivatives do.

This commit adds the following runtime IDs:
 - "miraclelinux"
 - "miraclelinux-x64"
 - "miraclelinux.8"
 - "miraclelinux.8-x64"
 - "miraclelinux.9"
 - "miraclelinux.9-x64"

bartonjs commented:
Seems OK to me (and CI is happy now).  Any other comments, @wfurt?

wfurt reviewed (approved):
LGTM

m10k commented:
Awesome, thank you everyone!
